configserver/src/main/java/com/yahoo/vespa/config/server/application/ReindexingStatus.java

```diff
@@ -0,0 +1,132 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.config.server.application;
+
+import com.yahoo.config.model.api.Reindexing;
+
+import java.time.Instant;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.toUnmodifiableMap;
+
+/**
+ * Pending and ready reindexing per document type. Each document type can have either a pending or a ready reindexing.
+ * This is immutable.
+ *
+ * @author jonmv
+ */
+public class ReindexingStatus implements Reindexing {
+
+    private static final ReindexingStatus empty = new ReindexingStatus(Map.of(), Map.of());
+
+    private final Map<String, Long> pending;
+    private final Map<String, Status> ready;
+
+    ReindexingStatus(Map<String, Long> pending, Map<String, Status> ready) {
+        this.pending = Map.copyOf(pending);
+        this.ready = ready;
```

Comment:
What is the reason for not copying the second parameter?

Reply:
Silly author, most likely >_<